From e6e9df368273f05f26fcafc5bae34369951c3ded Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@chainguard.dev>
Date: Tue, 29 Sep 2026 10:07:04 +0000
Subject: [PATCH] ssl: check ssl_context ofr null before setting options

If ssl_context creation fails, it will be NULL and the
SSL_CTX_set_options() call crash the application.

Thus move the null pointer check right after the attempt to create
context, before trying to use it for setting options.

(cherry picked from commit 2abd94445abdfec42d887aa0778384dc20525de1)
---
 src/core/tsi/ssl_transport_security.cc | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/core/tsi/ssl_transport_security.cc b/src/core/tsi/ssl_transport_security.cc
index 29ec47eeee..ed2bc1bd62 100644
--- a/src/core/tsi/ssl_transport_security.cc
+++ b/src/core/tsi/ssl_transport_security.cc
@@ -3309,15 +3309,15 @@ tsi_result tsi_create_ssl_client_handshaker_factory_with_options(
   ssl_context = SSL_CTX_new(TLS_method());
 #else
   ssl_context = SSL_CTX_new(TLSv1_2_method());
-#endif
-#if OPENSSL_VERSION_NUMBER >= 0x10101000 && !defined(LIBRESSL_VERSION_NUMBER)
-  SSL_CTX_set_options(ssl_context, SSL_OP_NO_RENEGOTIATION);
 #endif
   if (ssl_context == nullptr) {
     tsi::LogSslErrorStack();
     LOG(ERROR) << "Could not create ssl context.";
     return TSI_INVALID_ARGUMENT;
   }
+#if OPENSSL_VERSION_NUMBER >= 0x10101000 && !defined(LIBRESSL_VERSION_NUMBER)
+  SSL_CTX_set_options(ssl_context, SSL_OP_NO_RENEGOTIATION);
+#endif
 
   result = tsi_set_min_and_max_tls_versions(
       ssl_context, options->min_tls_version, options->max_tls_version);
@@ -3515,15 +3515,15 @@ tsi_result tsi_configure_server_ssl_context(
   ssl_context.ssl_ctx = SSL_CTX_new(TLS_method());
 #else
   ssl_context.ssl_ctx = SSL_CTX_new(TLSv1_2_method());
-#endif
-#if OPENSSL_VERSION_NUMBER >= 0x10101000 && !defined(LIBRESSL_VERSION_NUMBER)
-  SSL_CTX_set_options(ssl_context.ssl_ctx, SSL_OP_NO_RENEGOTIATION);
 #endif
   if (ssl_context.ssl_ctx == nullptr) {
     tsi::LogSslErrorStack();
     LOG(ERROR) << "Could not create ssl context.";
     return TSI_OUT_OF_RESOURCES;
   }
+#if OPENSSL_VERSION_NUMBER >= 0x10101000 && !defined(LIBRESSL_VERSION_NUMBER)
+  SSL_CTX_set_options(ssl_context.ssl_ctx, SSL_OP_NO_RENEGOTIATION);
+#endif
 
   tsi_result result = tsi_set_min_and_max_tls_versions(
       ssl_context.ssl_ctx, options->min_tls_version, options->max_tls_version);
-- 
2.53.0

